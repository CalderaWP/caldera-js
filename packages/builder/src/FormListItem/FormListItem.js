import React from 'react';
import { Toolbar, IconButton } from '@wordpress/components';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const FormListItemButton = ({
	icon,
	label,
	formId,
	actionName,
	onFormAction,
}) => (
	<IconButton
		icon={icon}
		label={label}
		className={`form-list-item form-list-item-${actionName}`}
		onClick={() => {
			onFormAction(formId, actionName);
		}}
	/>
);

FormListItemButton.propTypes = {
	icon: PropTypes.string,
	label: PropTypes.string.isRequired,
	formId: PropTypes.string.isRequired,
	actionName: PropTypes.string.isRequired,
};

FormListItemButton.defaultProps = {
	icon: 'edit',
};
/**
 *
 * @param form
 * @param onFormAction
 * @return {*}
 * @constructor
 */
export const FormListItem = ({ form, onFormAction, items }) => {
	const formId = form.id;
	return (
		<div>
			<div>{form.name}</div>
			<Toolbar>
				{items.map(item => (
					<FormListItemButton
						key={item.actionName}
						formId={formId}
						onFormAction={onFormAction}
						{...item}
					/>
				))}
			</Toolbar>
		</div>
	);
};
const items = [
	{
		icon: 'edit',
		label: 'Edit Form',
		actionName: 'edit',
	},
	{
		icon: 'list-view',
		label: 'View Entries',
		actionName: 'view-entries',
	},
	{
		icon: 'download',
		label: 'Export',
		actionName: 'export',
	},
	{
		icon: 'feedback',
		label: 'Preview Form',
		actionName: 'preview',
	},
	{
		icon: 'admin-settings',
		label: 'Form Settings',
		actionName: 'settings',
	},
];
FormListItem.defaultProps = {
	classNames: {
		edit: 'form-list-item-edit',
		'view-entries': 'form-list-item-view-entries',
		preview: 'form-list-item-preview',
		settings: 'form-list-item-settings',
	},
	items,
};
FormListItem.propTypes = {
	form: PropTypes.object,
	onFormAction: PropTypes.func.isRequired,
	items: PropTypes.array,
};
