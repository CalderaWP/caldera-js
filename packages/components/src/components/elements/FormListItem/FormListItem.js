import React from 'react';
import { Toolbar, IconButton } from '@wordpress/components';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 *
 * @param form
 * @param onFormAction
 * @return {*}
 * @constructor
 */
export const FormListItem = ({ form, onFormAction }) => (
	<div>
		<div>{form.name}</div>
		<Toolbar>
			<IconButton
				icon="edit"
				label="Edit Form"
				className={'form-list-item form-list-item-edit'}
				onClick={() => {
					onFormAction(form.id, 'edit');
				}}
			/>
			<IconButton
				icon="list-view"
				label="View Entries"
				className={'form-list-item form-list-item-view-entries'}
				onClick={() => {
					onFormAction(form.id, 'view-entries');
				}}
			/>
			<IconButton
				icon="feedback"
				label="Preview Form"
				className={'form-list-item form-list-item-preview'}
				onClick={() => {
					onFormAction(form.id, 'preview');
				}}
			/>
			<IconButton
				icon="admin-settings"
				label="Form Settings"
				className={'form-list-item form-list-item-settings'}
				onClick={() => {
					onFormAction(form.id, 'settings');
				}}
			/>
		</Toolbar>
	</div>
);

FormListItem.defaultProps = {
	classNames: {
		edit: 'form-list-item-edit',
		'view-entries': 'form-list-item-view-entries',
		preview: 'form-list-item-preview',
		settings: 'form-list-item-settings'
	}
};
FormListItem.propTypes = {
	form: PropTypes.object,
	onFormAction: PropTypes.func.isRequired
};
