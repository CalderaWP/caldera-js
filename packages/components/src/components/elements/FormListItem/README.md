# Forms List Item Component
Shows one item in the form list

## Usage
 
### Import With webpack
`import {FormsList}` from '@calderawp/components';`


### Render One Item
```jsx
	<FormListItem 
        form={forms['contact-form']} 
        onFormAction={() => {}} 
	/>

```

### Action Callback
The prop `onFormAction` passed to this component is called whenever any of the form action buttons are clicked. The callback will receive form ID and the action name -- edit,preview, etc.


```jsx
    const onFormAction = (formId,actionName) => {
        switch( actionName ){
            //...
        }
    });
	<FormListItem 
        form={forms['contact-form']} 
        onFormAction={onFormAction} 
	/>

```
