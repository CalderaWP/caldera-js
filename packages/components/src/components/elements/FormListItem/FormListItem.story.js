import React from 'react';
import { storiesOf } from '@storybook/react';
import { FormListItem } from './FormListItem';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { forms } from '../../EntryViewer/forms.fixtures';

storiesOf('FormListItem', module).add('A form in list', () => (
	<FormListItem form={forms['contact-form']} onFormAction={() => {}} />
));
