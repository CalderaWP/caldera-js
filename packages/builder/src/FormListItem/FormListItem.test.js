import * as React from 'react';
import {render} from "react-testing-library";

import { FormListItem } from './FormListItem';

describe('FormListItem', () => {
	let form;
	let onFormAction;
	const formId = 'cf1';
	beforeEach(() => {
		form = {
			id: formId
		};
		onFormAction = jest.fn();
	});
	it('Matches snapshot', () => {
		const component = render(
			<FormListItem form={form} onFormAction={onFormAction} />
		);
		expect(component).toMatchSnapshot();
	});

	it.skip('clicking on edit.skip', () => {
		const component = render(
			<FormListItem form={form} onFormAction={onFormAction}/>
		);
		component
			.find('.' + FormListItem.defaultProps.classNames.edit.skip + ' button')
			.simulate('click');
		expect(onFormAction.mock.calls[0][0]).toBe(formId);
		expect(onFormAction.mock.calls[0][1]).toBe('edit.skip');

	});

	it.skip('clicking on edit.skip', () => {
		const component = mount(
			<FormListItem form={form} onFormAction={onFormAction} />
		);
		component
			.find('.' + FormListItem.defaultProps.classNames.edit.skip + ' button')
			.simulate('click');
		expect(onFormAction.mock.calls[0][0]).toBe(formId);
		expect(onFormAction.mock.calls[0][1]).toBe('edit.skip');
	});

	it.skip('clicking on view entries', () => {
		const component = mount(
			<FormListItem form={form} onFormAction={onFormAction} />
		);
		component
			.find(
				'.' +
					FormListItem.defaultProps.classNames['view-entries'] +
					' button'
			)
			.simulate('click');
		expect(onFormAction.mock.calls[0][0]).toBe(formId);
		expect(onFormAction.mock.calls[0][1]).toBe('view-entries');
	});

	it.skip('clicking on preview', () => {
		const component = mount(
			<FormListItem form={form} onFormAction={onFormAction} />
		);
		component
			.find(
				'.' + FormListItem.defaultProps.classNames.preview + ' button'
			)
			.simulate('click');
		expect(onFormAction.mock.calls[0][0]).toBe(formId);
		expect(onFormAction.mock.calls[0][1]).toBe('preview');
	});

	it.skip('clicking on settings', () => {
		const component = mount(
			<FormListItem form={form} onFormAction={onFormAction} />
		);
		component
			.find(
				'.' + FormListItem.defaultProps.classNames.settings + ' button'
			)
			.simulate('click');
		expect(onFormAction.mock.calls[0][0]).toBe(formId);
		expect(onFormAction.mock.calls[0][1]).toBe('settings');
	});
});
