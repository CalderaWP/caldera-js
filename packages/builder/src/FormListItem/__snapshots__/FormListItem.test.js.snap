// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormListItem Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div>
        <div />
        <div
          class="components-toolbar"
        >
          <button
            aria-label="Edit Form"
            class="components-button components-icon-button form-list-item form-list-item-edit"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="dashicon dashicons-edit"
              focusable="false"
              height="20"
              role="img"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.89 3.39l2.71 2.72c.46.46.42 1.24.03 1.64l-8.01 8.02-5.56 1.16 1.16-5.58s7.6-7.63 7.99-8.03c.39-.39 1.22-.39 1.68.07zm-2.73 2.79l-5.59 5.61 1.11 1.11 5.54-5.65zm-2.97 8.23l5.58-5.6-1.07-1.08-5.59 5.6z"
              />
            </svg>
          </button>
          <button
            aria-label="View Entries"
            class="components-button components-icon-button form-list-item form-list-item-view-entries"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="dashicon dashicons-list-view"
              focusable="false"
              height="20"
              role="img"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 19h16c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1H2c-.55 0-1 .45-1 1v16c0 .55.45 1 1 1zM4 3c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V3h11zM4 7c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V7h11zM4 11c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11zM4 15c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11z"
              />
            </svg>
          </button>
          <button
            aria-label="Export"
            class="components-button components-icon-button form-list-item form-list-item-export"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="dashicon dashicons-download"
              focusable="false"
              height="20"
              role="img"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.01 4v6h2V2H4v8h2.01V4h8zm-2 2v6h3l-5 6-5-6h3V6h4z"
              />
            </svg>
          </button>
          <button
            aria-label="Preview Form"
            class="components-button components-icon-button form-list-item form-list-item-preview"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="dashicon dashicons-feedback"
              focusable="false"
              height="20"
              role="img"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 2h16c.55 0 1 .45 1 1v14c0 .55-.45 1-1 1H2c-.55 0-1-.45-1-1V3c0-.55.45-1 1-1zm15 14V7H3v9h14zM4 8v1h3V8H4zm4 0v3h8V8H8zm-4 4v1h3v-1H4zm4 0v3h8v-3H8z"
              />
            </svg>
          </button>
          <button
            aria-label="Form Settings"
            class="components-button components-icon-button form-list-item form-list-item-settings"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="dashicon dashicons-admin-settings"
              focusable="false"
              height="20"
              role="img"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 16V4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h13c.55 0 1-.45 1-1zM8 11h1c.55 0 1 .45 1 1s-.45 1-1 1H8v1.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V13H6c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V11zm5-2h-1c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V7h1c.55 0 1 .45 1 1s-.45 1-1 1h-1v5.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V9z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div />
      <div
        class="components-toolbar"
      >
        <button
          aria-label="Edit Form"
          class="components-button components-icon-button form-list-item form-list-item-edit"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-edit"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.89 3.39l2.71 2.72c.46.46.42 1.24.03 1.64l-8.01 8.02-5.56 1.16 1.16-5.58s7.6-7.63 7.99-8.03c.39-.39 1.22-.39 1.68.07zm-2.73 2.79l-5.59 5.61 1.11 1.11 5.54-5.65zm-2.97 8.23l5.58-5.6-1.07-1.08-5.59 5.6z"
            />
          </svg>
        </button>
        <button
          aria-label="View Entries"
          class="components-button components-icon-button form-list-item form-list-item-view-entries"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-list-view"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 19h16c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1H2c-.55 0-1 .45-1 1v16c0 .55.45 1 1 1zM4 3c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V3h11zM4 7c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6V7h11zM4 11c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11zM4 15c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm13 0v2H6v-2h11z"
            />
          </svg>
        </button>
        <button
          aria-label="Export"
          class="components-button components-icon-button form-list-item form-list-item-export"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-download"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.01 4v6h2V2H4v8h2.01V4h8zm-2 2v6h3l-5 6-5-6h3V6h4z"
            />
          </svg>
        </button>
        <button
          aria-label="Preview Form"
          class="components-button components-icon-button form-list-item form-list-item-preview"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-feedback"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 2h16c.55 0 1 .45 1 1v14c0 .55-.45 1-1 1H2c-.55 0-1-.45-1-1V3c0-.55.45-1 1-1zm15 14V7H3v9h14zM4 8v1h3V8H4zm4 0v3h8V8H8zm-4 4v1h3v-1H4zm4 0v3h8v-3H8z"
            />
          </svg>
        </button>
        <button
          aria-label="Form Settings"
          class="components-button components-icon-button form-list-item form-list-item-settings"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="dashicon dashicons-admin-settings"
            focusable="false"
            height="20"
            role="img"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 16V4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h13c.55 0 1-.45 1-1zM8 11h1c.55 0 1 .45 1 1s-.45 1-1 1H8v1.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V13H6c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V11zm5-2h-1c-.55 0-1-.45-1-1s.45-1 1-1h1V5.5c0-.28.22-.5.5-.5s.5.22.5.5V7h1c.55 0 1 .45 1 1s-.45 1-1 1h-1v5.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5V9z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
